### PR TITLE
CSmartnodesync deadlocks fix + static analyze fixes + initial seednod…

### PR DESCRIPTION
**Pull request by malbit**
…es fetch repair

### CODE REVIEW

File: src/smartnode/smartnode-sync.cpp [hunk 1]
```diff
@@ -239,37 +246,10 @@ void CSmartnodeSync::ProcessTick(CConnman& connman)
...
-                            // reset nTimeNoObjectsLeft to be able to use the same condition on resync
-                            nTimeNoObjectsLeft = 0;
-                            SwitchToNextAsset(connman);
-                            connman.ReleaseNodeVector(vNodesCopy);
-                            return;
-                        }
-                        nLastTick = nTick;
-                        nLastVotes = governance.GetVoteCount();
-                    }
+                    // will request votes on per-obj basis from each node in a separate loop
```
Comment (CodeIsTheKey) [inline src/smartnode/smartnode-sync.cpp:251]: Nice catch

File: src/smartnode/smartnode-sync.cpp [hunk 2]
```diff
@@ -324,10 +335,15 @@ void CSmartnodeSync::NotifyHeaderTip(const CBlockIndex *pindexNew, bool fInitial
 void CSmartnodeSync::UpdatedBlockTip(const CBlockIndex *pindexNew, bool fInitialDownload, CConnman& connman)
 {
     LogPrint(BCLog::MNSYNC, "CSmartnodeSync::UpdatedBlockTip -- pindexNew->nHeight: %d fInitialDownload=%d\n", pindexNew->nHeight, fInitialDownload);
-
     nTimeLastUpdateBlockTip = GetAdjustedTime();
 
-    if (IsSynced() || !pindexBestHeader)
+    CBlockIndex* pindexTip;
+    {
+        LOCK(cs_main);
```
Comment (CodeIsTheKey) [inline src/smartnode/smartnode-sync.cpp:342]: Nice

**Review by CodeIsTheKey — APPROVED** [review]
LGTM - nice cleanup and a couple good fixes.

**Review by npq7721 — APPROVED** [review]
LGTM

**Pull request merged**